### DISCUSSION TITLE
Is there a way to replace clipboard contents instead of adding to it?

### DISCUSSION THREAD

**Discussion by Tudmotu** [nvim-tree/nvim-tree.lua#2397] [Q&A] [answered]
NvimTree has the feature where you can add multiple files to your "clipboard" then paste them all at once.
While this is a cool feature, it sometimes gets in my way since I press "c" or "x" on some file by accident, then I copy/cut the actual file I wanted, but when I paste it it pastes both files.

Is there a way to configure NvimTree to *replace* the contents of the clipboard on every `c`/`x` press?
Or alternatively, is there a way to clear the clipboard?

Thanks

**Comment by alex-courtis** [comment, marked as answer]
The clipboard is additive, without an option to replace.

You can clear the clipboard `:help nvim-tree-api.fs.clear_clipboard()`

Create a function:
```lua
local function clear_and_copy()
  api.fs.clear_clipboard()
  api.fs.copy.node()
end
```

Map it in your on_attach:
```lua
vim.keymap.set('n', 'c',     clear_and_copy,                      opts('Clear And Copy'))
```

**Reply by alex-courtis** [reply]
- [x] @alex-courtis add to recipe if this works

**Reply by Tudmotu** [reply]
Thank you!
Will that only work inside the NvimTree?

**Reply by alex-courtis** [reply]
Mappings using the "standard" on_attach opts `:help nvim-tree-mappings` only apply inside the nvim-tree buffer.

You can apply them everywhere by removing `bufnr` however I don't recommend it. I don't think I'm understanding your use case...

**Reply by Tudmotu** [reply]
Thanks! It works :smiley: 
Regarding buffer - thanks, I understand it now. I only want it inside NvimTree.
This is the `on_attach` function I used:
```lua
on_attach = function (bufnr)
    local api = require "nvim-tree.api"

    local function opts(desc)
      return { desc = "nvim-tree: " .. desc, buffer = bufnr, noremap = true, silent = true, nowait = true }
    end

    -- default mappings
    api.config.mappings.default_on_attach(bufnr)
    local function clear_and_copy()
        api.fs.clear_clipboard()
        api.fs.copy.node()
    end

    local function clear_and_cut()
        api.fs.clear_clipboard()
        api.fs.cut()
    end

    vim.keymap.set('n', 'c', clear_and_copy, opts('Clear And Copy'))
    vim.keymap.set('n', 'x', clear_and_cut, opts('Clear And Cut'))
end,
```

**Reply by alex-courtis** [reply]
That's great.

I'd be most grateful if you documented your [Recipe](https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes)

**Reply by Tudmotu** [reply]
Added:
https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes#clear-clipboard-before-every-copy--cut-operation

**Reply by alex-courtis** [reply]
Many thanks!
